1.9.0-rc1 : Traceback in libkv/store/boltdb/boltdb.go on delete of non-existent key

```
panic: runtime error: slice bounds out of range

goroutine 77 [running]:
github.com/docker/libkv/store/boltdb.func·006(0xc208055860, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libkv/store/boltdb/boltdb.go:333 +0x204
github.com/boltdb/bolt.(*DB).Update(0xc208963080, 0xc208a3f6e8, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/boltdb/bolt/db.go:459 +0xde
github.com/docker/libkv/store/boltdb.(*BoltDB).AtomicDelete(0xc208711bd0, 0xc2086eea80, 0x5d, 0xc208a85350, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libkv/store/boltdb/boltdb.go:339 +0x1da
github.com/docker/libnetwork/datastore.(*datastore).DeleteObjectAtomic(0xc20876a7b0, 0x7fa6f0125ae8, 0xc208a8e0a0, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/datastore/datastore.go:476 +0x368
github.com/docker/libnetwork.(*controller).deleteFromStore(0xc208714bd0, 0x7fa6f0125ae8, 0xc208a8e0a0, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/store.go:190 +0x2c0
github.com/docker/libnetwork.(*sandbox).storeDelete(0xc2087ff6c0, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/sandbox_store.go:147 +0x15e
github.com/docker/libnetwork.(*sandbox).Delete(0xc2087ff6c0, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/docker/libnetwork/sandbox.go:172 +0x14e
[...]
```

--- a/Godeps/Godeps.json
+++ b/Godeps/Godeps.json
@@ -156,7 +156,7 @@
 		},
 		{
 			"ImportPath": "github.com/docker/libkv",
-			"Rev": "749af6c5b3fb755bec1738cc5e0d3a6f1574d730"
+			"Rev": "c2aac5dbbaa5c872211edea7c0f32b3bd67e7410"
 		},
 		{
 			"ImportPath": "github.com/godbus/dbus",

--- a/Godeps/_workspace/src/github.com/docker/libkv/store/boltdb/boltdb.go
+++ b/Godeps/_workspace/src/github.com/docker/libkv/store/boltdb/boltdb.go
@@ -330,6 +330,9 @@ func (b *BoltDB) AtomicDelete(key string, previous *store.KVPair) (bool, error) 
 		}
 
 		val = bucket.Get([]byte(key))
+		if val == nil {
+			return store.ErrKeyNotFound
+		}
 		dbIndex := binary.LittleEndian.Uint64(val[:libkvmetadatalen])
 		if dbIndex != previous.LastIndex {
 			return store.ErrKeyModified

--- a/Godeps/_workspace/src/github.com/docker/libkv/store/consul/consul.go
+++ b/Godeps/_workspace/src/github.com/docker/libkv/store/consul/consul.go
@@ -467,6 +467,13 @@ func (s *Consul) AtomicDelete(key string, previous *store.KVPair) (bool, error) 
 	}
 
 	p := &api.KVPair{Key: s.normalize(key), ModifyIndex: previous.LastIndex}
+
+	// Extra Get operation to check on the key
+	_, err := s.Get(key)
+	if err != nil && err == store.ErrKeyNotFound {
+		return false, err
+	}
+
 	if work, _, err := s.client.KV().DeleteCAS(p, nil); err != nil {
 		return false, err
 	} else if !work {

--- a/Godeps/_workspace/src/github.com/docker/libkv/store/etcd/etcd.go
+++ b/Godeps/_workspace/src/github.com/docker/libkv/store/etcd/etcd.go
@@ -368,6 +368,10 @@ func (s *Etcd) AtomicDelete(key string, previous *store.KVPair) (bool, error) {
 	_, err := s.client.Delete(context.Background(), s.normalize(key), delOpts)
 	if err != nil {
 		if etcdError, ok := err.(etcd.Error); ok {
+			// Key Not Found
+			if etcdError.Code == etcd.ErrorCodeKeyNotFound {
+				return false, store.ErrKeyNotFound
+			}
 			// Compare failed
 			if etcdError.Code == etcd.ErrorCodeTestFailed {
 				return false, store.ErrKeyModified

--- a/Godeps/_workspace/src/github.com/docker/libkv/store/zookeeper/zookeeper.go
+++ b/Godeps/_workspace/src/github.com/docker/libkv/store/zookeeper/zookeeper.go
@@ -347,9 +347,15 @@ func (s *Zookeeper) AtomicDelete(key string, previous *store.KVPair) (bool, erro
 
 	err := s.client.Delete(s.normalize(key), int32(previous.LastIndex))
 	if err != nil {
+		// Key not found
+		if err == zk.ErrNoNode {
+			return false, store.ErrKeyNotFound
+		}
+		// Compare failed
 		if err == zk.ErrBadVersion {
 			return false, store.ErrKeyModified
 		}
+		// General store error
 		return false, err
 	}
 	return true, nil

--- a/Godeps/_workspace/src/github.com/docker/libkv/testutils/utils.go
+++ b/Godeps/_workspace/src/github.com/docker/libkv/testutils/utils.go
@@ -312,6 +312,11 @@ func testAtomicDelete(t *testing.T, kv store.Store) {
 	success, err = kv.AtomicDelete(key, pair)
 	assert.NoError(t, err)
 	assert.True(t, success)
+
+	// Delete a non-existent key; should fail
+	success, err = kv.AtomicDelete(key, pair)
+	assert.Error(t, store.ErrKeyNotFound)
+	assert.False(t, success)
 }
 
 func testLockUnlock(t *testing.T, kv store.Store) {

